witness/src/Dscp/Witness/Logic/Getters.hs

```diff
@@ -36,21 +37,33 @@ import Dscp.Witness.Launcher.Context
 import Dscp.Witness.Logic.Exceptions
 import Dscp.Witness.Mempool ()
 import Dscp.Witness.SDLock
+import Dscp.Witness.Web.Types (BlocksOrMempool (..))
+
+-- | Run an action both in chain-only and chain+mempool modes.
+runSdDual
+    :: WitnessWorkMode ctx m
+    => (forall mode. KnownSdReadMode mode => SdReadM mode m a)
+    -> m (BlocksOrMempool a)
+runSdDual action =
+    sequence BlocksOrMempool
+    { bmConfirmed = runSdReadM @'ChainOnly action
+    , bmTotal = runSdReadM @'ChainAndMempool action
+    }
```

Comment:
Seems like a very bad helper which should never be used (as it would cause double query to blockchain)

Reply:
Mm yeah, it's probably worth removing this function.
Although doing two requests is what snowdrop assumes as to do at the moment.

Reply:
@Martoon-00 you claim that we cannot query _just_ the mempool in Snowdrop and there is no built-in way of obtaining such a dual result via a single query?

Reply:
Well, AFAIS snowdrop's way of accessing data allows only one exact source of data (either only chain or chain with mempool) per `runERoComp` (similar to our `runSdM`) run. 
Need to ask Gosha about that.

Reply:
We discussed this with @Heimdell and came to conclusion, that this dual db access is probably ok due to caching, need benchmarks to judge about it. And snowdrop indeed does not allow to do one db access from the code point of view.

Reply:
Confirming. At least different states will be in different AVL+ trees. Also, rocksdb has a very good cache and is cheap to read.

Reply:
Cool then!